server/app/controllers/BlocksController.scala

```diff
@@ -41,15 +41,35 @@ class BlocksController @Inject()(
       .toFuture
   }
 
-  def getBlockHeader(blockhashString: String) = public { _ =>
-    blockService
-      .getBlockHeader(blockhashString)
-      .toFutureOr
-      .map { value =>
-        val response = Ok(Json.toJson(value))
-        response.withHeaders("Cache-Control" -> "public, max-age=31536000")
-      }
-      .toFuture
+  /**
+   * Try to retrieve a blockHeader by height, in case the query argument
+   * is not a valid height, we assume it might be a blockhash and try to
+   * retrieve the blockHeader by blockhash.
+   */
+  def getBlockHeader(query: String) = public { _ =>
+    Try(query.toInt)
+      .map(Height.apply)
+      .map(
```

Comment:
When using an anonymous function that spans over several lines, it is common to use braces instead of parenthesis, example:
```
     Try(query.toInt)
      .map(Height.apply)
      .map {  value =>
          blockService
            .getBlockHeader(value)
            .toFutureOr
            .map { value =>
              val response = Ok(Json.toJson(value))
              response.withHeaders("Cache-Control" -> "no-store")
            }
            .toFuture
      }
```